### PR TITLE
Expire token on email change

### PR DESCRIPTION
**Pull request by acinader**
fixes: #5104

### CODE REVIEW

File: src/Routers/ClassesRouter.js [hunk 1]
```diff
@@ -114,7 +125,7 @@ export class ClassesRouter extends PromiseRouter {
       where,
       req.body,
       req.info.clientSDK
-    );
+    ).then(this.afterUpdate.bind(this, req));
```
Comment (acinader) [inline src/Routers/ClassesRouter.js]: there may be a much more natural place to put this...I was hoping in the UserRouter, but the update handler there doesn't actually get called.  I don't understand exactly why that is, but the fact is that this is the handler that gets a called....

File: src/Controllers/UserController.js [hunk 2]
```diff
@@ -242,21 +242,26 @@ export class UserController extends AdaptableController {
...
-            {
-              _perishable_token: { __op: 'Delete' },
-              _perishable_token_expires_at: { __op: 'Delete' },
-            }
-          )
-        )
+        .then(user =>
+          Promise.all([
+            updateUserPassword(user.objectId, password, this.config),
+            this.clearPasswordResetToken(user.objectId)
```
Comment (flovilmart) [inline src/Controllers/UserController.js]: Can we do it all in one write?

File: src/Routers/ClassesRouter.js (hunk 1 above)
Comment (flovilmart) [reply src/Routers/ClassesRouter.js]: I’m not sure why we need to duplicate the logic here. User handling should be done in one spot don’t you think?

File: src/Controllers/UserController.js (hunk 2 above)
Comment (acinader) [reply src/Controllers/UserController.js]: yes

**Comment by acinader** [comment]
yup, that's MUCH better.  I like this now.

**Comment by flovilmart** [comment]
Way better indeed

File: src/RestWrite.js [hunk 3]
```diff
@@ -95,6 +95,9 @@ RestWrite.prototype.execute = function() {
     .then(() => {
       return this.runBeforeTrigger();
     })
+    .then(() => {
+      return this.deleteEmailRestTokenIfNeeded();
+    })
```
Comment (acinader) [inline src/RestWrite.js]: Doh, typo.  Fix me, should be Reset token, not Rest

Comment (flovilmart) [reply src/RestWrite.js]: Ohh noooo

**Pull request merged**